src/west/main.py

```diff
@@ -506,13 +507,23 @@ def main(argv=None):
     except WestUpdated:
         # West has been automatically updated. Restart ourselves to run the
         # latest version, with the same arguments that we were given.
-        log.dbg("sys.executable:\"{}\" sys.argv[0]:\"{}\" argv:\"{}\"".format(
-                sys.executable, sys.argv[0], argv))
-        if sys.platform == "win32":
-            # On Windows Python generates .exe executables
-            os.execv(sys.argv[0], argv)
-        else:
-            os.execv(sys.executable, [sys.executable] + [sys.argv[0]] + argv)
+        # Treat the Python script as an executable. This works because on
+        # Unix the script created by pip has a shebang and on Windows it is
+        # actually a binary executable
+        log.dbg("sys.argv[0]:\"{}\" argv:\"{}\"".format(sys.argv[0], argv))
+        # Use Popen + exit instead of execv due to the asynchronous nature of
+        # execv # on Windows, where it creates a new process with a different
```

Comment:
Looks like these got badly reflowed, note # after execv, and other lines too.

Reply:
Fixed